sandbox/Devis/Keyvault.tf

```
@@ -0,0 +1,32 @@
+resource "azurerm_key_vault" "kv_azure_demo_01" {
+  lifecycle {
+    prevent_destroy = true
+  }
+
+  name                = "kv-azure-demo-01"
+  resource_group_name = azurerm_resource_group.sandbox.name
+  location            = azurerm_resource_group.sandbox.location
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+  tags                = local.default_tags
+}
+
+resource "azurerm_key_vault_secret" "username" {
+  key_vault_id = azurerm_key_vault.kv_azure_demo_01.id
+  name         = "username"
+  value        = "azure_demo_user"
+}
+
+resource "random_string" "random_key_generator" {
+  length  = 32
+  lower   = true
+  numeric = true
+  upper   = false
+  special = true
```

In password generation, recommended to set below parameters as well.. specially limiting to only acceptable special chars based on where password will be used
```suggestion
  min_lower        = 2
  min_numeric      = 2
  min_special      = 2
  min_upper        = 2
  override_special = "!#$@"
```